Controllers/userController.js

```diff
@@ -0,0 +1,248 @@
+  const checktoken = await User.findOne({ token });
+  if (!checktoken) {
+    const message = 'Incorrect Token! Register again';
+    return sendError(res, [], message);
+  }
+
+  //Find matching user for the token
+  const user = await User.findOne({ _id: checktoken._id });
+  if (!user) {
+    const message = 'Error Occured';
+    return sendError(res, [], message);
+  }
+
+  //Check if user had already been verified
+  if (user.isVerified) {
+    const message = 'This user has alredy been verified';
+    return sendSuccess(res, [], message);
+  }
+
+  //Else, setIsverified to true and token to empty
+  const setIsverified = await User.updateMany({ isVerified: true, token: '' });
+  if (!setIsverified) {
+    const message = 'Internal error! Try again';
+    return sendError(res, [], message);
+  }
+  const message = 'Registration Success';
+  return sendSuccess(res, [], message);
+};
+
+const forgotPasswordController = async (req, res) => {
+  const { email } = req.body;
+
+  //CHECK IF EMAIL EXIST IN THE DATABASE
+  const user = await User.findOne({ email });
+  if (!user) {
+    const message = 'User with this email does not exist';
+    return sendError(res, [], message);
+  }
+
+  // CREATE AND ASSIGN A TOKEN
+  const token = jwt.sign({ _id: user._id }, process.env.FORGOT_PASSWORD_TOKEN, {
+    expiresIn: '20m',
+  });
+
+  // CREATE EMAIL PASSWORD RESET MESSAGE
+  const link = `<div>
+
+      <p>Click this link to reset password</>
+
+      <p>${process.env.CLIENT_URL}/api/resetpassword/${token}</p>
+
+      <p>All The Best</p>
+    </div>
+    `;
+
+  //EMAIL CONFIGURATION
+  const mg = await mailgun({
+    apiKey: process.env.MAILGUN_API_KEY,
+    domain: process.env.MAILGUN_DOMAIN,
+  });
+
+  const data = {
+    from: '"youngprinnce 👻" <' + process.env.EMAIL_ADDRESS + '>', // sender address
+    to: `${email}`, // list of receivers
+    subject: 'Password rest Link ✔', // Subject line
+    text: 'hello',
+    html: link, // plain text body
+  };
+
+  const resetLink = await User.updateOne({ resetLink: token });
+  if (!resetLink) {
+    const message = 'reset password link error';
+    return sendError(res, [], message);
+  }
+
+  try {
+    await mg.messages().send(data);
+    const message = `Password reset link has been sent to ${email}`;
+    return sendSuccess(res, [], message);
+  } catch (error) {
+    const message = 'reset password link error';
+    return sendError(res, err, message);
+  }
+};
+
+const resetPasswordController = async (req, res) => {
+  //Retrive token from URL
+  const token = req.params.token;
+  const password = req.body.password;
+
+  //Verify token
+  const verified = await jwt.verify(token, process.env.FORGOT_PASSWORD_TOKEN);
+  if (!verified) {
+    const message = 'Incorrect or Expired Link! Register Again';
+    return sendError(res, [], message);
+  }
+
+  if (!verified) {
+    const message = 'Incorrect or Expired Link! Register Again';
+    return sendError(res, [], message);
+  }
```

Comment:
This is duplicate, the two lines are the same

Reply:
Mistake!!, didn't see that